### ISSUE TITLE
java.lang.IllegalStateException: Expected statement but was LABEL_NAME..

### ISSUE THREAD

**Issue by cSchobi** [google/closure-compiler#3946]
When compiling the following javascript code:
```
(function () {
    label1:
        label2:
            try {
            } catch (e) {
            }
});
```

I get the following stacktrace:

```
java.lang.IllegalStateException: Expected statement but was LABEL_NAME.. Reference node:
LABEL_NAME label1 2:4  [length: 6] [source_file: delta_js_smallest.js]

 Parent node:
BLOCK 1:13  [length: 89] [source_file: delta_js_smallest.js]
    LABEL_NAME label1 2:4  [length: 6] [source_file: delta_js_smallest.js]

	at com.google.javascript.jscomp.AstValidator$1.handleViolation(AstValidator.java:86)
	at com.google.javascript.jscomp.AstValidator.violation(AstValidator.java:2112)
	at com.google.javascript.jscomp.AstValidator.validateStatement(AstValidator.java:258)
	at com.google.javascript.jscomp.AstValidator.validateStatement(AstValidator.java:155)
	at com.google.javascript.jscomp.AstValidator.validateBlock(AstValidator.java:979)
	at com.google.javascript.jscomp.AstValidator.validateFunctionBody(AstValidator.java:1127)
	at com.google.javascript.jscomp.AstValidator.validateFunctionExpressionHelper(AstValidator.java:1103)
	at com.google.javascript.jscomp.AstValidator.validateFunctionExpression(AstValidator.java:1078)
	at com.google.javascript.jscomp.AstValidator.validateExpression(AstValidator.java:430)
	at com.google.javascript.jscomp.AstValidator.validateExprStmt(AstValidator.java:1538)
	at com.google.javascript.jscomp.AstValidator.validateStatement(AstValidator.java:211)
	at com.google.javascript.jscomp.AstValidator.validateStatement(AstValidator.java:155)
	at com.google.javascript.jscomp.AstValidator.validateStatements(AstValidator.java:149)
	at com.google.javascript.jscomp.AstValidator.validateScript(AstValidator.java:138)
	at com.google.javascript.jscomp.AstValidator.validateCodeRoot(AstValidator.java:124)
	at com.google.javascript.jscomp.AstValidator.process(AstValidator.java:109)
	at com.google.javascript.jscomp.PhaseOptimizer$NamedPass.process(PhaseOptimizer.java:267)
	at com.google.javascript.jscomp.PhaseOptimizer.process(PhaseOptimizer.java:186)
	at com.google.javascript.jscomp.Compiler.performFinalizations(Compiler.java:2584)
	at com.google.javascript.jscomp.Compiler.lambda$stage3Passes$6(Compiler.java:959)
	at com.google.javascript.jscomp.CompilerExecutor.lambda$runInCompilerThread$0(CompilerExecutor.java:101)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(Thread.java:829)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:597)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:194)
```
I used version v20220502
(Note: This bug occurs on a larger javascript function, the example above is simplified)

Steps to reproduce:
* Run `google-closure-compiler --js delta_js_smallest.js` where `delta_js_smalles.js` contains the javascript code above.

**Comment by concavelenz** [comment]
Thanks for the report. I was able to reproduce this.

**Comment by concavelenz** [comment]
Should be fixed in b15ce7c4f90ff6919c679ca7f05cee9eed2bac87
